### PR TITLE
Added sleep timer

### PR DESCRIPTION
**Pull request by rasmuslos**

### CODE REVIEW

File: ios/App/App/plugins/AbsAudioPlayer.swift [hunk 1]
```diff
@@ -95,13 +99,69 @@ public class AbsAudioPlayer: CAPPlugin {
     }
     
     @objc func sendMetadata() {
-        self.notifyListeners("onPlayingUpdate", data: [ "value": !PlayerHandler.paused() ])
+        self.notifyListeners("onPlayingUpdate", data: [ "value": !PlayerHandler.paused ])
```
Comment (advplyr) [inline ios/App/App/plugins/AbsAudioPlayer.swift:102]: I get an error here without (). I see you removed the (), is this a version difference?

Comment (advplyr) [reply ios/App/App/plugins/AbsAudioPlayer.swift:102]: I see now you made them all getters & setters but my xcode doesn't recognize that. Maybe I need to update xcode or something

Comment (rasmuslos) [reply ios/App/App/plugins/AbsAudioPlayer.swift:102]: > I get an error here without (). I see you removed the (), is this a version difference?

Sorry, I don't know how I missed that. It has to be `.paused`

> I see now you made them all getters & setters but my xcode doesn't recognize that. Maybe I need to update xcode or something

That is really weird. I am on macOS `12.3.1 (21E258)` and Xcode `13.3.1 (13E500a)`

Comment (advplyr) [reply ios/App/App/plugins/AbsAudioPlayer.swift:102]: It is working today, not sure what happened.

File: ios/App/App/plugins/AbsAudioPlayer.swift [hunk 2]
```diff
@@ -95,13 +99,69 @@ public class AbsAudioPlayer: CAPPlugin {
...
+        
+        PlayerHandler.remainingSleepTime = currentSleepTime - time
+        call.resolve()
+    }
+    @objc func increaseSleepTime(_ call: CAPPluginCall) {
+        guard let timeString = call.getString("time") else { return call.resolve([ "success": false ]) }
+        guard let time = Int(timeString) else { return call.resolve([ "success": false ]) }
+        guard let currentSleepTime = PlayerHandler.remainingSleepTime else { return call.resolve([ "success": false ]) }
+        
+        PlayerHandler.remainingSleepTime = currentSleepTime + time
```
Comment (advplyr) [inline ios/App/App/plugins/AbsAudioPlayer.swift:122]: Time is coming in milliseconds. We need to change to
`PlayerHandler.remainingSleepTime = currentSleepTime + (time / 1000)`

**Comment by advplyr** [comment]
Nice!

**Pull request merged**